pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -206,6 +206,13 @@ func (a *acrProvider) Provide() credentialprovider.DockerConfig {
 			cfg[url] = *cred
 		}
 	}
+
+	// add ACR anonymous repo support: use empty username and password for anonymous access
+	cfg["*.azurecr.*"] = credentialprovider.DockerConfigEntry{
```

Comment:
What all does this match? does this match "foo.azurecr.bar.com" ?

Reply:
@sivagms Yes, it match, but I think it does not matter since if user use private repo, suppose they should provide pull image secret by themselves.

Reply:
That’s a bit scary. Some malicious user can just change the registry name in the chart to receive the AKS SP and password

Reply:
currently the matching patterns are like following:
( *. azurecr.io, SP) ( *. azurecr.cn, SP)  ( * . azurecr .* , empty password)
Per offline discussion, it's safe
Thanks for the good point!